### ISSUE TITLE
UI API table is either broken/removed, or not named "UI"

### ISSUE THREAD

**Issue by Neocloudy** [Sceleratis/Adonis#1163]
TLDR: the UI table cannot be accessed in v240/v239 even if the key is correct and access is enabled. UI will return nil and warn that the access is disabled or the key is incorrect. Probably broken/removed or just not named `UI` (***why***)

- This bug is occurring on the Release Github branch of Adonis, v240/v239 [Tested with both the loader on Roblox and release branch of repo], all the time if you attempt to call the UI table properly.

- Expected behavior is for the UI API table to return accordingly, even if your settings allow API access to the UI table and the function you use to call the table uses the correct key. Without returning nil and warning that Access is disabled or the key is incorrect.

- If you attempt to call the `UI` API table, even with UI calls enabled, the access key set properly, `Access` and `API` both enabled, it will return nil and warn that the access key is incorrect or access is disabled.

- This bug impacts being able to access the UI API table.

- Seems to be an Adonis feature that broke at some point. On a version that worked in 2020/2021, calling UI worked. I've tested on v239 and the release branch [v240] and it does not work anymore.

- note that the repeated warnings are just the script i wrote attempting to try again as it still returns nil

Purgatory. Table calls tested with both a one-line one attempt script to access the UI table, and a function that automatically handles retrying if UI returns nil. Also tested with no plugins, with write access set, in Studio and in a regular server, etc, and it continues returning nil. And, of course, the other API tables I attempted to call [Remote and Admin] were returning properly.

Also, the white lines that read :: API :: are not from Adonis. They're from the script used to try accessing them. this still does not matter as I have tried both a simple method and this function method.
![image](https://github.com/Epix-Incorporated/Adonis/assets/88008002/d21f4b85-efc2-4c0a-8156-c04295c12c92)
![image](https://github.com/Epix-Incorporated/Adonis/assets/88008002/cd2a67cf-adb3-4cae-b443-2a09608b82e7)

**Comment by watameln** [comment]
Ui is meant for client is it not?

**Comment by karl-police** [comment]
> TLDR: the UI table cannot be accessed in v240/v239 even if the key is correct and access is enabled. UI will return nil and warn that the access is disabled or the key is incorrect. Probably broken/removed or just not named `UI` (_**why**_)
> 
> * This bug is occurring on the Release Github branch of Adonis, v240/v239 [Tested with both the loader on Roblox and release branch of repo], all the time if you attempt to call the UI table properly.
> * Expected behavior is for the UI API table to return accordingly, even if your settings allow API access to the UI table and the function you use to call the table uses the correct key. Without returning nil and warning that Access is disabled or the key is incorrect.
> * If you attempt to call the `UI` API table, even with UI calls enabled, the access key set properly, `Access` and `API` both enabled, it will return nil and warn that the access key is incorrect or access is disabled.
> * This bug impacts being able to access the UI API table.
> * Seems to be an Adonis feature that broke at some point. On a version that worked in 2020/2021, calling UI worked. I've tested on v239 and the release branch [v240] and it does not work anymore.
> * note that the repeated warnings are just the script i wrote attempting to try again as it still returns nil
> 
> Purgatory. Table calls tested with both a one-line one attempt script to access the UI table, and a function that automatically handles retrying if UI returns nil. Also tested with no plugins, with write access set, in Studio and in a regular server, etc, and it continues returning nil. And, of course, the other API tables I attempted to call [Remote and Admin] were returning properly.
> 
> Also, the white lines that read :: API :: are not from Adonis. They're from the script used to try accessing them. this still does not matter as I have tried both a simple method and this function method. ![image](https://user-images.githubusercontent.com/88008002/253034089-d21f4b85-efc2-4c0a-8156-c04295c12c92.png) ![image](https://user-images.githubusercontent.com/88008002/253034108-cd2a67cf-adb3-4cae-b443-2a09608b82e7.png)

How are you accessing the API?

Can you show a line of code?

&nbsp;
From what I see is it will tell you that the access is disabled, if you're doing it from the Client, and Client is not set to true.

**Comment by ccuser44** [comment]
UI only exists on the client to my knowledge so that's why it's probably not possible
